File: codility/perm_miss_elem.cc
// 100%
// x^x = 0; 0^x = x
int solution(vector<int> &A) {
    int n = A.size();
    int xor_sum = 0;
    
    for (int i = 0; i < n; i++) {
        xor_sum = xor_sum ^ A[i] ^ (i+1);
    }
    return xor_sum ^ (n+1);
}

File: archive/codility/lesson_1_done/perm_miss_elem.py
# 100% both
def solution(A):
    n = len(A)+1
    sa = sum(A)
    sn = n*(n+1)/2
    return sn - sa

# http://codesays.com/2014/solution-to-perm-missing-elem-by-codility/ 
def solution(A):
    n = len(A)
    k = 0 
    for i in range(n):
        k = k ^ A[i] ^ (i+1)
    return k ^ (n+1)